plugins/vst_base/RemoteVstPlugin.cpp

```diff
@@ -1068,6 +1081,45 @@ const char * RemoteVstPlugin::programName()
 
 
 
+// join the ParameterDisplays (stringified values without units) and send them to host
+void RemoteVstPlugin::getParameterDisplays()
+{
+	std::string presName;
+	static char buf[9]; // buffer for getting string
+	for (int i=0; i< m_plugin->numParams; ++i)
+	{
+		memset( buf, 0, sizeof( buf ) ); // fill with '\0' because got string may not to be ended with '\0'
+		pluginDispatch( effGetParamDisplay, i, 0, buf );
+		buf[8] = 0;
+		if (i != 0) presName += '|';
```

Comment:
I think it's unsafe to concatenate names with a specific separator because some plugin may use it in the parameter name. There are at least three safe ways:
- Escape the name of the parameter
- Send each name separately
- Prepend the length of the string at the beginning and don't use any separators, ex. `0-2-AM2--14-Gain`